### PR TITLE
Remove dependency pinnings from Gemfile

### PR DESCRIPTION
**Pull request by ghoneycutt**
Attempt to simplify the Gemfile by removing unnecessary version pinning.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,17 +7,17 @@ else
...
-  gem 'puppetlabs_spec_helper', '>= 2.11.0',              :require => false
-  gem 'puppet-lint', "~> 2.0",                            :require => false
-  gem 'json', "~> 1.8.3",                                 :require => false
-  gem 'json_pure', "~> 1.8.3",                            :require => false
+  gem 'puppetlabs_spec_helper',                           :require => false
   gem 'metadata-json-lint',                               :require => false
   gem 'puppet-blacksmith',                                :require => false
-  gem 'rest-client', ">= 1.7.4",                          :require => false
+  gem 'json',                                             :require => false
+  gem 'json_pure',                                        :require => false
```
Comment (treydock) [inline Gemfile]: We don't directly use `json` or `json_pure`, they should probably be pulled in by whatever else needs them.  Maybe remove them?  I think same could be said for `rest-client`.

File: Gemfile [hunk 2]
```diff
@@ -41,18 +41,18 @@ group :documentation do
 end
 
 group :system_tests do
-  gem 'beaker', '~> 4.x',             :require => false
```
Comment (treydock) [inline Gemfile]: This would only become a problem if/when beaker 5 comes out, the 3 to 4 transition wasn't painful but required changes to testing.

Comment (ghoneycutt) [reply Gemfile]: We'll find that during the automated tests and can always pin if needed.

File: Gemfile (hunk 1 above)
Comment (ghoneycutt) [reply Gemfile]: Let's do this in another PR after this one passes

**Pull request merged**